src/yunohost/app.py

```diff
@@ -156,10 +156,18 @@ def app_info(app, full=False):
 
 
 def _app_upgradable(app_infos):
+    from packaging import version
 
     # Determine upgradability
     # In case there is neither update_time nor install_time, we assume the app can/has to be upgraded
 
+    # Firstly use the version to know if an upgrade is available
+    if app_infos["version"] != "-" and app_infos["from_catalog"]["manifest"].get("version", None):
```

Comment:
Are we sure that it's gonna be a "good" version if it's not "-" though ? Couldn't it be something messy that's not "-" ?

Also I was initially thinking about checking if "upgrade_only_if_version_changes" is set before doing this check ... But that's debatable

Reply:
> Also I was initially thinking about checking if "upgrade_only_if_version_changes" is set before doing this check ... But that's debatable

Yes it's better to check this before.

Reply:
Should be down